browser-test/src/support/admin_programs.ts

```diff
@@ -577,25 +587,64 @@ export class AdminPrograms {
       await this.addQuestionFromQuestionBank(questionName)
     }
   }
-  async editProgramBlockWithBlockName(
+
+  /**
+   * Edit basic block details and required and optional questions. Cannot handle more than one optional question.
+   * @deprecated prefer using {@link #editProgramBlockUsingSpec} instead.
+   */
+  async editProgramBlockWithOptional(
     programName: string,
-    blockName: string,
     blockDescription = 'screen description',
-    questionNames: string[] = [],
+    questionNames: string[],
+    optionalQuestionName: string,
   ) {
     await this.gotoEditDraftProgramPage(programName)
 
     await clickAndWaitForModal(this.page, 'block-description-modal')
-    await this.page.fill('#block-name-input', blockName)
     await this.page.fill('textarea', blockDescription)
-    // Make sure input validation enables the button before clicking.
     await this.page.click('#update-block-button:not([disabled])')
 
+    // Add the optional question
+    await this.addQuestionFromQuestionBank(optionalQuestionName)
+    // Only allow one optional question per block; this selector will always toggle the first optional button.  It
+    // cannot tell the difference between multiple option buttons
+    await this.page.click(`:is(button:has-text("optional"))`)
```

Comment:
Not retrofitting old methods to support multiple options. If multiple are needed people should swap to the new method `editProgramBlockUsingSpec`